pkg/api/apiloader.go

```diff
@@ -205,6 +205,9 @@ func (a *Apiloader) LoadContainerService(
 		if containerService.Properties == nil {
 			return nil, errors.New("missing ContainerService Properties")
 		}
+		if containerService.Properties.IsAzureStackCloud() && containerService.Location == "" {
```

Comment:
I don't think this is required. Location is a mandatory parameter and we will throw an error even if a user does not mention it.

Reply:
@tariq1890 

Yes, it is required for AzureStack.
During executing the generate command, the location is used to construct the FQDN as alternative DNS into apiserver.crt.
For Azure, the locations are per-defined list and do not need the location information in container service.  alternative DNS in apiserver.crt will include all azure supported locations.
See the example at the bottom

So the location could be missing from api model file and template and certificate are correct for generate command.

For Azure Stack, the location is retrieved from location filed of container service.  So the location filed need to be present in api model file. Otherwise the alternative DNS in apiserver.crt will be incorrect (missing the location part) and the template are generated by generate command will not work for Azure Stack. 
So I add the check to ensure the location is present in api model file.



Alternative DNS in apiserver.crt for Azure

X509v3 Subject Alternative Name:
                DNS:k111007.australiacentral.cloudapp.azure.com, DNS:k111007.australiacentral2.cloudapp.azure.com, DNS:k111007.australiaeast.cloudapp.azure.com, DNS:k111007.australiasoutheast.cloudapp.azure.com, DNS:k111007.brazilsouth.cloudapp.azure.com, DNS:k111007.canadacentral.cloudapp.azure.com, DNS:k111007.canadaeast.cloudapp.azure.com, DNS:k111007.centralindia.cloudapp.azure.com, DNS:k111007.centralus.cloudapp.azure.com, DNS:k111007.centraluseuap.cloudapp.azure.com, DNS:k111007.chinaeast.cloudapp.chinacloudapi.cn, DNS:k111007.chinaeast2.cloudapp.chinacloudapi.cn, DNS:k111007.chinanorth.cloudapp.chinacloudapi.cn, DNS:k111007.chinanorth2.cloudapp.chinacloudapi.cn, DNS:k111007.eastasia.cloudapp.azure.com, DNS:k111007.eastus.cloudapp.azure.com, DNS:k111007.eastus2.cloudapp.azure.com, DNS:k111007.eastus2euap.cloudapp.azure.com, DNS:k111007.francecentral.cloudapp.azure.com, DNS:k111007.francesouth.cloudapp.azure.com, DNS:k111007.japaneast.cloudapp.azure.com, DNS:k111007.japanwest.cloudapp.azure.com, DNS:k111007.koreacentral.cloudapp.azure.com, DNS:k111007.koreasouth.cloudapp.azure.com, DNS:k111007.northcentralus.cloudapp.azure.com, DNS:k111007.northeurope.cloudapp.azure.com, DNS:k111007.southcentralus.cloudapp.azure.com, DNS:k111007.southeastasia.cloudapp.azure.com, DNS:k111007.southindia.cloudapp.azure.com, DNS:k111007.uksouth.cloudapp.azure.com, DNS:k111007.ukwest.cloudapp.azure.com, DNS:k111007.westcentralus.cloudapp.azure.com, DNS:k111007.westeurope.cloudapp.azure.com, DNS:k111007.westindia.cloudapp.azure.com, DNS:k111007.westus.cloudapp.azure.com, DNS:k111007.westus2.cloudapp.azure.com, DNS:k111007.chinaeast.cloudapp.chinacloudapi.cn, DNS:k111007.chinanorth.cloudapp.chinacloudapi.cn, DNS:k111007.chinanorth2.cloudapp.chinacloudapi.cn, DNS:k111007.chinaeast2.cloudapp.chinacloudapi.cn, DNS:k111007.germanycentral.cloudapp.microsoftazure.de, DNS:k111007.germanynortheast.cloudapp.microsoftazure.de, DNS:k111007.usgovvirginia.cloudapp.usgovcloudapi.net, DNS:k111007.usgoviowa.cloudapp.usgovcloudapi.net, DNS:k111007.usgovarizona.cloudapp.usgovcloudapi.net, DNS:k111007.usgovtexas.cloudapp.usgovcloudapi.net, DNS:k111007.francecentral.cloudapp.azure.com, DNS:localhost, DNS:kubernetes, DNS:kubernetes.default, DNS:kubernetes.default.svc, DNS:kubernetes.default.svc.cluster.local, DNS:kubernetes.kube-system, DNS:kubernetes.kube-system.svc, DNS:kubernetes.kube-system.svc.cluster.local, IP Address:10.240.255.5, IP Address:127.0.0.1, IP Address:10.240.255.15, IP Address:10.240.255.6, IP Address:10.240.255.7, IP Address:10.0.0.1